pkg/traceql/lexer.go

```diff
@@ -95,30 +99,48 @@ func (l *lexer) Lex(lval *yySymType) int {
 		return END_ATTRIBUTE
 	}
 
-	r := l.Scan()
-
 	// if we are currently parsing an attribute then just grab everything until we find a character that ends the attribute.
 	// we will handle parsing this out in ast.go
 	if l.parsingAttribute {
-		str := l.TokenText()
-		// parse out any scopes here
-		tok := tokens[str+string(l.Peek())]
+		// parse out any scopes here, but copy it so we dont move main scanner if it's not a scope
```

Comment:
is this section complex enough to move to its own function? wdyt?

Reply:
yup, I will make the change

Reply:
Done, have refactored the attribute parsing piece to separate function